src/main/java/gregtech/api/metatileentity/SyncedTileEntityBase.java

```diff
@@ -26,13 +25,30 @@ public abstract class SyncedTileEntityBase extends BlockStateTileEntity {
 
     public abstract void receiveCustomData(int discriminator, PacketBuffer buf);
 
-    protected final Int2ObjectMap<byte[]> updates = new Int2ObjectArrayMap<>(5);
+    private final PacketDataList updates = new PacketDataList();
 
     public void writeCustomData(int discriminator, Consumer<PacketBuffer> dataWriter) {
         ByteBuf backedBuffer = Unpooled.buffer();
         dataWriter.accept(new PacketBuffer(backedBuffer));
         byte[] updateData = Arrays.copyOfRange(backedBuffer.array(), 0, backedBuffer.writerIndex());
-        updates.put(discriminator, updateData);
+        this.updates.add(discriminator, updateData);
+        if (this.updates.size() == 1) notifyWorld(); // if the data is not empty we already notified the world
```

Comment:
Should this check `if (this.updates.size()) > 0`?

Reply:
No, we only want to notify if the updates were empty. Since we just added one we need to check `size == 1`. The comment might be missleading.